LookUp Event Handling: rewrite (#1172)
DevExpress/devextreme-demos: JSDemos/Demos/Lookup/EventHandling/description.md

> To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.

Reviewer: Where should I specify the mentioned function?

> To handle the Lookup component's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.

Reviewer 2: ```suggestion
To handle changes to the Lookup component's value, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display a picture and text for the selected item.
```

> The Lookup applies the selected value instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*. This functionality is not used in this demo.

Reviewer: > If you want users to click the Apply button instead

This is not an end in itself. The thing is that the described option allows you to delay the effects of your change until a click on the "Apply" button.

> This functionality **is not used** in this demo.

This passive voice is totally avoidable from the look of it. For example:

> This demo does not show this functionality.

> The Lookup applies the selection instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*. In this demo, you can use a switch to toggle this property.

Reviewer 3: > In this demo, you can use a switch to toggle this property.

When will this functionality be added to the demo?

```suggestion
The Lookup applies the selected value instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*. In this demo, you can use a switch to toggle this property.
```

> To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) handler. This demo uses it to display the picture and text for the selected item.

Reviewer 4: ```suggestion
To handle the Lookup's value change, specify the [onValueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#onValueChanged) function. This demo uses it to display the picture and text for the selected item.
```

If you want to change event handler at runtime or attach several handlers to the 

> [valueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Events/#valueChanged) event, use the [on(eventName, eventHandler)](/Documentation/ApiReference/UI_Components/dxLookup/Methods/#oneventName_eventHandler) method to subscribe to this event. This approach is more typical of jQuery. For Angular and Vue, you can use two-way data bindings. To do this in React, create a state variable.

Reviewer 4: Just a reminder - you can create a separate description for each approach. Vue developers are not interested in React, Angular devs in jQuery and so on.

> If you want to change event handler at runtime or attach several handlers to the 
> [valueChanged](/Documentation/ApiReference/UI_Components/dxLookup/Events/#valueChanged) event, use the [on(eventName, eventHandler)](/Documentation/ApiReference/UI_Components/dxLookup/Methods/#oneventName_eventHandler) method to subscribe to this event. This approach is more typical of jQuery. For Angular and Vue, you can use two-way data bindings. To do this in React, create a state variable.

Reviewer 3: The scenario with `on` is rare. Two-way data bindings are not about event handling. Let's remove this paragraph entirely.

```suggestion
```

> The Lookup applies the selection instantly. If you want users to click the Apply button instead, set the [applyValueMode](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#applyValueMode) property to *"useButtons"*.

Reviewer 3: We haven't mentioned selection yet, so this paragraph looks out of place. Why do we even mention this property? It's not in the demo, and it is related to the topic only barely.

> To specify the DOM events after which search results should be updated, use the 

Reviewer 3: "search results"? Isn't `valueChangeEvent` specify an event on which to the change the `value` property?

> To specify the DOM events after which search results should be updated, use the 
> [valueChangeEvent](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#valueChangeEvent) property. The default events are input, change, and keyup.

Reviewer 4: ```suggestion
To specify the <a href="https://developer.mozilla.org/en-US/docs/Web/Events" target="_blank">DOM events</a> after which search results should be updated, use the [valueChangeEvent](/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#valueChangeEvent) property. The default events are *"input"*, *"change"*, and *"keyup"*.
```
